### PR TITLE
insights handler deployment

### PR DESCRIPTION
**Pull request by timclifford**
corresponding PR uselagoon/lagoon#2997

### CODE REVIEW

**Comment by bomoko** [comment]
@timclifford - this LGTM, but I'd like someone with more Helm-foo to look at it too.

File: charts/lagoon-core/values.yaml [hunk 1]
```diff
@@ -465,6 +465,24 @@ backupHandler:
...
     # targetMemoryUtilizationPercentage: 80
 
+insightsHandler:
+  enabled: true
+  replicaCount: 2
+  image:
+    repository: uselagoon/insights-handler
+    pullPolicy: Always
+    # Overrides the image tag whose default is the chart appVersion.
+    tag: ""
```
Comment (tobybellwood) [inline charts/lagoon-core/values.yaml:475]:
```suggestion
    repository: testlagoon/insights-handler
    pullPolicy: Always
    # Overrides the image tag whose default is the chart appVersion.
    tag: "pr-2997"
```

Comment (tobybellwood) [reply charts/lagoon-core/values.yaml:475]: If we do it this way, we can test that this works, and then once the corresponding Lagoon release has come out, we can revert this commit

**Comment by tobybellwood** [comment]
we need to be able to control the setting for the `s3-host` to be used from the chart - same as the logs2s3 service

**Comment by tobybellwood** [comment]
THe insights-handler service doesn't expose a port 3000, so the liverness and readiness checks don't work - we can either disable the checks, or expose a port (unnecessarily?)

File: charts/lagoon-core/templates/insights-handler.deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        - name: RABBITMQ_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ include "lagoon-core.broker.fullname" . }}
+              key: RABBITMQ_PASSWORD
+        - name: S3_FILES_BUCKET
+          value: {{ required "A valid .Values.s3FilesBucket required!" .Values.s3FilesBucket | quote }}
+        - name: S3_FILES_HOST
+          value: {{ required "A valid .Values.s3FilesHost required!" .Values.s3FilesHost | quote }}
+        {{- with .Values.s3AccessKeyID }}
```
Comment (tobybellwood) [inline charts/lagoon-core/templates/insights-handler.deployment.yaml:52]:
```suggestion
        {{- with .Values.s3FilesAccessKeyID }}
```

File: charts/lagoon-core/templates/insights-handler.deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        - name: S3_FILES_ACCESS_KEY_ID
+          value: {{ . | quote }}
+        {{- end }}
+        {{- with .Values.s3SecretAccessKey }}
```
Comment (tobybellwood) [inline charts/lagoon-core/templates/insights-handler.deployment.yaml:56]:
```suggestion
        {{- with .Values.s3FilesSecretAccessKey }}
```

**Comment by tobybellwood** [comment]
ok - I've modified this to:
* disable insights-handler (for now) in both the linter and the test-suite
* point it to the uselagoon/insights-handler:main image
* leave it available to use downstream, by enabling it in helm values

**Pull request merged**